cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -0,0 +1,893 @@
+                                                    (genesisUtxo Map.! 0) -- corresponds to 'genesisAddress'.
+                                                    targetAddress
+
+  let ((_, _), signingKey) = genesisUtxo Map.! 0
+      outBig = assumeBound $ CC.Common.mkLovelace 863000000000000
+      outForBig = CC.UTxO.TxOut
+        { CC.UTxO.txOutAddress = targetAddress
+        , CC.UTxO.txOutValue   = outBig
+        }
+
+  let genesisTx :: GenTx (ByronBlockOrEBB ByronConfig)
+      genesisTx = txSpendGenesisUTxOByronPBFT genesisConfig
+                                              signingKey
+                                              genesisAddress
+                                              (NE.fromList [outForBig])
+
+  submitTx cc pInfoConfig (node topologyInfo) genesisTx llTracer
+
+  -- Done, the first transaction 'initGenTx' is submitted,
+  -- now 'sourceAddress' has a lot of money.
+  case mFunds of
+    Nothing             -> return ()
+    Just (txInIndex, _) -> do
+      let txIn  = CC.UTxO.TxInUtxo (Byron.byronTxId genesisTx) txInIndex
+          txOut = outForBig
+      addToAvailableFunds (txIn, txOut)
+    -- Now we can use these money for further transactions.
+
+-- | Take the initial funds from a single richhombre to a specific address.
+extractInitialFunds :: NodeConfig ByronEBBExtNodeConfig
+                    -> (TxDetails, Crypto.SigningKey)
+                    -> CC.Common.Address                    -- the address to send the funds
+                    -> ( Maybe (Word32, CC.Common.Lovelace) -- the index and value of 'funds'
+                       , CC.Common.Lovelace                 -- the associated fees
+                       , GenTx (ByronBlockOrEBB cfg)
+                       )
+extractInitialFunds cfg input address =
+  (funds, fees, initGenTx)
+ where
+  (funds, fees, _, initGenTx) =
+    mkTransaction cfg
+                  input
+                  (Just address)
+                  (Set.empty :: Set (Int, CC.UTxO.TxOut))
+
+-- | Single input to multiple (in sequence) output transaction (special case for benchmarking).
+mkTransaction
+  :: (FiscalRecipient r)
+  => NodeConfig ByronEBBExtNodeConfig
+  -> (TxDetails, Crypto.SigningKey)       -- TxIn, TxOut that will be used as
+                                          -- input and the key to spend the
+                                          -- associated value
+  -> Maybe CC.Common.Address              -- the address to associate with the 'change',
+                                          -- if different from that of the first argument
+  -> Set (r, CC.UTxO.TxOut)               -- each recipient and their payment details
+  -> ( Maybe (Word32, CC.Common.Lovelace) -- the 'change' index and value (if any)
+     , CC.Common.Lovelace                 -- the associated fees
+     , Map r Word32                       -- the offset map in the transaction below
+     , GenTx (ByronBlockOrEBB cfg)
+     )
```

Comment:
Not a big deal but, typically, once I start dealing with anything more than a triple, I like to define a separate data type.